packages/sit-onyx/src/components/OnyxDataGrid/features/filtering/filtering.ts

```diff
@@ -0,0 +1,141 @@
+export const useFiltering = createFeature(
+  <TEntry extends DataGridEntry>(options?: FilterOptions<TEntry>) => {
+    const filters = ref({}) as Ref<FilterState<DataGridEntry>>;
+    const { t } = injectI18n();
+    const config = computed(() => toValue(options?.columns));
+
+    const isFilterEnabled = computed(() => (col: keyof TEntry) => {
+      return config.value?.[col]?.enabled !== false;
+    });
+
+    const filterData = (entries: Readonly<TEntry>[]) => {
+      return entries.filter((entry) =>
+        Object.entries(filters.value).every(
+          ([column, value]: [keyof TEntry, string | undefined]) => {
+            const columnOptions = config.value?.[column];
+            const filterOptions = { ...options?.filterConfig, ...columnOptions?.config };
+
+            if (value == null || value === "") return true;
+            let searchTerm = value.toString();
+            let entryValue = entry[column]?.toString() ?? "";
+            if (filterOptions?.filterFunc) {
+              return filterOptions.filterFunc(
+                searchTerm,
+                entryValue as TEntry[keyof TEntry],
+                column,
+                entry,
+              );
+            }
+            entryValue = removeDiacritics(entryValue);
+            searchTerm = removeDiacritics(searchTerm);
+            if (!filterOptions.caseSensitive) {
+              entryValue = entryValue.toLowerCase();
+              searchTerm = searchTerm.toLowerCase();
+            }
+            if (filterOptions.trimWhitespace) {
+              entryValue = entryValue.replace(/\s+/g, "");
+            }
+            if (filterOptions.searchFromStart) {
+              return entryValue.startsWith(searchTerm);
+            }
+            if (filterOptions.exactMatch) {
+              return entryValue === searchTerm;
+            }
+
+            return entryValue.includes(searchTerm);
+          },
+        ),
+      );
+    };
+
+    const clearFilter = (column: keyof DataGridEntry) => {
+      filters.value[column] = "";
+    };
+
+    // sync filters with user provided config
+    watchEffect(() => {
+      if (!config.value) return;
+      Object.entries(config.value).forEach(([column, columnOptions]) => {
+        if (columnOptions?.searchTerm) {
+          filters.value[column] = columnOptions.searchTerm;
+        }
+      });
+    });
+
+    const getMenuItem = (column: keyof DataGridEntry) => {
+      let inputValue = filters.value[column] || "";
+
+      const handleKeyDown = (e: KeyboardEvent) => {
+        if (e.key === "Enter") {
+          filters.value[column] = inputValue;
+        }
+      };
+
+      return h(OnyxMiniSearch, {
+        label: t.value("dataGrid.head.filtering.menu.label", { column: column.toString() }),
+        class: "onyx-filter-search",
+        placeholder: t.value(`dataGrid.head.filtering.menu.placeholder`),
+        modelValue: inputValue,
+        autofocus: true,
```

Comment:
autofocus does not seem to work. When opening the flyout, the input is not focused so the user has to explicitly click again to be able to search